cirq/aqt/aqt_sampler.py

```diff
@@ -0,0 +1,224 @@
+    """Sampler for the AQT ion trap device
+    This sampler connects to the AQT machine and
+    runs a single circuit or an entire sweep remotely
+    """
+
+    def __init__(self, remote_host: str, access_token: str):
+        """
+        Args:
+            remote_host: Address of the remote device.
+            access_token: Access token for the remote api.
+        """
+        self.remote_host = remote_host
+        self.access_token = access_token
+
+    def _run_api(
+            self,
+            circuit: circuits.Circuit,
+            param_resolver: Sweepable,
+    ) -> str:
+        """
+        Args:
+            circuit: Circuit to be run
+            param_resolver: Param resolver for the
+
+        Returns:
+            json formatted string of the sequence
+        """
+
+        #seq_list: List[Tuple[str, float, List[int]]] = []
+        seq_list = []
+        circuit = resolve_parameters(circuit, param_resolver)  # type: ignore
+        # TODO: Check if circuit is empty
+        for op in circuit.all_operations():
+            qubits = [obj.x for obj in op.qubits]  # type: ignore
+            op_str = get_op_string(op.gate)  # type: ignore
+            seq_list.append((op_str, op.gate.exponent, qubits))  # type: ignore
+        json_list = json.dumps(seq_list)
+        return json_list
+
+    def _send_json(self,
+                   *,
+                   json_str: str,
+                   id_str: Union[str, uuid.UUID],
+                   repetitions: int = 1,
+                   num_qubits: int = 1):
+        """Sends the json string to the remote AQT device
+        Args:
+            json_str: Json representation of the circuit.
+            id_str: Unique id of the datapoint.
+            repetitions: Number of repetitions.
+            num_qubits: Number of qubits present in the device.
+
+        Returns:
+            Measurement results as an array of boolean.
+        """
+        while True:
+            time.sleep(1.0)
+            data = put(self.remote_host,
+                       data={
+                           'data': json_str,
+                           'id': id_str,
+                           'acccess_token': self.access_token,
+                           'repetitions': repetitions,
+                           'num_qubits': num_qubits
+                       }).json()
+            if data['status'] == 'finished':
+                break
+        measurements_int = data['samples']
+        measurements = np.zeros((len(measurements_int), num_qubits))
+        for i, result_int in enumerate(measurements_int):
+            for j in range(num_qubits):
+                measurements[i, j] = np.floor(result_int / 2**j)
+        return measurements
+
+    def run_sweep(self,
+                  program: Union[circuits.Circuit, schedules.Schedule],
+                  params: study.Sweepable,
+                  repetitions: int = 1,
+                  num_qubits: int = 1) -> List[study.TrialResult]:
```

Comment:
Shouldn't `num_qubits` be computed from the circuit or schedule? Why does it have to be specified?